### PR TITLE
Issue #3197706 by rolki: As a CM I want to be able to set join methods for the course

### PR DESCRIPTION
**Pull request by rollki**
**Problem/Motivation**
There is no way to select join methods for the course

**Steps to reproduce**
When you go to the group edit page then you should be able to select join methods for the course

**Proposed resolution**
Extend Group Request feature

**Tickets**
- https://www.drupal.org/project/social_course/issues/3197706
- https://getopensocial.atlassian.net/browse/YANG-4740
- https://github.com/goalgorilla/open_social/pull/2227

### CODE REVIEW

File: modules/social_course_basic/modules/social_course_basic_request/config/install/field.field.group.course_basic.field_group_allowed_join_method.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    - field.storage.group.field_group_allowed_join_method
+    - group.type.course_basic
+  module:
+    - options
+id: group.course_basic.field_group_allowed_join_method
+field_name: field_group_allowed_join_method
+entity_type: group
+bundle: course_basic
+label: 'Allowed join method'
+description: 'Choose how people can join this group.'
```
Comment (ribel) [inline modules/social_course_basic/modules/social_course_basic_request/config/install/field.field.group.course_basic.field_group_allowed_join_method.yml:14]: @rollki please rename `group` to `course` here and in all other places visible to the user.

**Review by ribel — CHANGES_REQUESTED** [review]
@rollki, In general, look good, just rename labels from group to course in string visible to the users.

File: modules/social_course_advanced/modules/social_course_advanced_request/social_course_advanced_request.info.yml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+name: 'Social Advanced Course Join'
```
Comment (ribel) [inline modules/social_course_advanced/modules/social_course_advanced_request/social_course_advanced_request.info.yml:1]: @rollki rename Join to Request here to be consistent.

File: modules/social_course_basic/modules/social_course_basic_request/config/install/field.field.group.course_basic.field_group_allowed_join_method.yml (hunk 1 above)
Comment (rollki) [reply modules/social_course_basic/modules/social_course_basic_request/config/install/field.field.group.course_basic.field_group_allowed_join_method.yml]: Done.

File: modules/social_course_advanced/modules/social_course_advanced_request/social_course_advanced_request.info.yml (hunk 2 above)
Comment (rollki) [reply modules/social_course_advanced/modules/social_course_advanced_request/social_course_advanced_request.info.yml]: Done.

**Review by ribel — APPROVED** [review]

**Review by ribel — APPROVED** [review]

**Pull request merged**